src/module.c

```diff
@@ -2486,6 +2506,342 @@ RedisModuleString *RM_ListPop(RedisModuleKey *key, int where) {
     return decoded;
 }
 
+/* Returns the element at index `index` in the list stored at `key`, like the
+ * LINDEX command.
+ *
+ * The index is zero-based, so 0 means the first element, 1 the second element
+ * and so on. Negative indices can be used to designate elements starting at the
+ * tail of the list. Here, -1 means the last element, -2 means the penultimate
+ * and so forth.
+ *
+ * When no value is found at the given key and index, an NULL is returned and
+ * `errno` is set as follows:
+ *
+ * - EINVAL if key is NULL.
+ * - ENOTSUP if the key is not a list.
+ * - EBADF if the key is not opened for reading.
+ * - EDOM if the index is out of range.
+ */
+RedisModuleString *RM_ListGet(RedisModuleKey *key, long index) {
+    if (!key) {
+        errno = EINVAL;
+        return NULL;
```

Comment:
i'm pretty sure everyone's gonna hate me for suggesting this, and hate me even more if it's merged, so maybe just as a joke that i can't resist:
Since i hate wasting screen real-estate (lines) on boilerplate error handling, and love the beauty of C (every statement returns a value), 
this is the short form (note there's no need for else-if if you return):
```suggestion
    if (!key) return errno = EINVAL, NULL;
```
@yossigo?

Reply:
I like it. Comma is an underestimated operator.

Reply:
i couldn't agree more. but i think we're a minority.

Reply:
@oranagra when it comes to code style I tend to think that "standard is better than better", and I don't think this is very common trick. But I don't feel strongly against that, so please don't count me as -1 (and @zuiderkwast is +1 apparently).